modules/benchmark/R/metric.residual.plot.R

```diff
@@ -8,21 +8,24 @@
 metric.residual.plot <- function(dat, var){
   
   require(ggplot2)
+  # 
+  # ind <- intersect(which(!is.na(dat$obvs)),  which(!is.na(dat$model)))
+  # 
+  # dat <- dat[ind,]
   
-  ind <- intersect(which(!is.na(dat$obvs)),  which(!is.na(dat$model)))
-  
-  dat <- dat[ind,]
-  dat$time <- lubridate::year(as.Date(as.character(dat$time), format="%Y",))
-  dat$diff <- dat$model - dat$obvs
+  dat$time <- lubridate::year(as.Date(as.character(dat$time), format="%Y"))
+  dat$diff <- abs(dat$model - dat$obvs)
+  dat$zeros <- rep(0, length(time))
   
-  ggplot(data = dat) + 
-    geom_path(aes(x=time,y=rep(0, length(time))), colour = "#666666", size=2, linetype = 2, lineend = "round") +
-    geom_point(aes(x=time,y=diff), size=4,  colour = "#619CFF") + labs(title=var, x= "years", y="model - observation")
+  p <- ggplot(data = dat, aes(x=time)) + 
+    geom_path(aes(y=zeros), colour = "#666666", size=2, linetype = 2, lineend = "round") +
+    geom_point(aes(y=diff), size=4,  colour = "#619CFF") + labs(title=var, x= "years", y="abs(model - observation)")
   
+  plot(p)
   # ind <- intersect(which(!is.na(dat$obvs)),  which(!is.na(dat$model)))
   # plot(dat$model[ind]-dat$obvs[ind], ylim = c(-max(dat$model[ind]-dat$obvs[ind]),max(dat$model[ind]-dat$obvs[ind])))
   # abline(h=0)
-  
+
   return(NA)
```

Comment:
I have a slightly different question. My understanding was that metric functions returned either scalar skill scores or pointers to files that we could insert into dbfiles and then have those point to the correct benchmarks_ensembles_scores. I'm fine with NA or NULL being the value that we insert into the skill score to indicate that a metric returns a file not a scalar, but this function doesn't seem to generate a file or return a pointer to said file.